adapters/final_characters_controller_test.go

```diff
@@ -0,0 +1,205 @@
+package adapters
+
+import (
```

Comment:
When ordering, it's more common to have this structure:

```go
import (
  // native packages
  // external
  // external packages - same organization
  // internal packages
)
```

All of the groups should be separated by a blank line.

Reply:
Done